genesyscloud/provider/division.go

```diff
@@ -0,0 +1,31 @@
+package provider
```

Comment:
Please review - This is refactored provider code.  Please review.  I had to duplicate this code from another resource to make sure we could deal with some cyclical dependencies